src/compiler/scanner.ts

```diff
@@ -337,6 +337,9 @@ namespace ts {
             Debug.assert(res < lineStarts[line + 1]);
         }
         else if (debugText !== undefined) {
+            if (res === debugText.length) {
+                return res; // Allow single character overflow for trailing newline
+            }
             Debug.assert(res < debugText.length);

```

Comment:
Couldn't you just use `<=` for the same effect? Currently this says: If `res` is `len`, return `res`, else assert that `res < len` and return `res` anyway.

Reply:
Yes; this just made it more explicit and less like an unintentional off-by-one error.